synapse/handlers/room.py

```diff
@@ -848,7 +859,9 @@ async def _send_events_for_new_room(
             The stream_id of the last event persisted.
         """
 
-        creator_id = creator.user.to_string()
+        if not creator:
+            creator = requester.user
+        creator_id = creator.to_string()
```

Comment:
I think mypy is complaining because it isn't sure what `request.user` is (if I add `reveal_type(request.user)` and then run mypy it reports `Any`).

The following seems equivalent and mypy seems OK with it:

```suggestion
        if not creator:
            creator_id = requester.user.to_string()
        else:
            creator_id = creator.to_string()
```

Reply:
This is not the same, `creator` is used again later in the code.